backend/src/main/kotlin/com/tngtech/apicenter/backend/config/Configuration.kt

```diff
@@ -1,17 +1,36 @@
 package com.tngtech.apicenter.backend.config
 
+import com.atlassian.crowd.service.client.CrowdClient
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.dataformat.yaml.YAMLMapper
 import com.fasterxml.jackson.module.kotlin.KotlinModule
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.beans.factory.annotation.Autowired
+import com.tngtech.apicenter.backend.connector.crowd.service.CrowdAuthenticationService
+import com.tngtech.apicenter.backend.connector.crowd.service.NullAuthenticationService
+import com.tngtech.apicenter.backend.domain.service.ExternalAuthenticationService
 
 @Configuration
-class Configuration {
+class Configuration @Autowired constructor(private val crowdClient: CrowdClient) {
```

Comment:
You don't need to do it like this. Methods that are `@Bean` annotated are automatically autowired.

You can write
```
@Bean
fun externalAuthenticationService(crowdClient: CrowdClient) {
...
}
```

But: Now you require even the users without Crowd to have this dependency.

After some googling, I think [ConditionalOnProperty](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/autoconfigure/condition/ConditionalOnProperty.html) would be preferable. What do you say?